### PR TITLE
Allow rpcallowip field to be configurable

### PR DESCRIPTION
**Pull request by Leibniz137**
~~- 0.23 was recently released~~
- i have found it useful to allow connections from my local LAN when my dappnode-wifi or vpn isn't working reliably. The env var was already set up for use in the entrypoint script, it just needed to be exposed in the UI

### CODE REVIEW

File: dappnode_package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "bitcoin.dnp.dappnode.eth",
-  "version": "0.1.6",
-  "upstreamVersion": "v22.0",
+  "version": "0.2.0",
```
Comment (alexpeterson91) [inline dappnode_package.json:3]: revert so we can apply SemVer and do a proper version update in CI

File: docker-compose.yml [hunk 2]
```diff
@@ -16,6 +16,7 @@ services:
       - BTC_TXINDEX=1
       - BTC_PRUNE=0
       - BTC_DISABLEWALLET=1
+      - BTC_RPCALLOWIP=172.33.0.0/16
```
Comment (alexpeterson91) [inline docker-compose.yml:19]: Gonna need more context about why you want this implemented since AFAIK this will not work for your desires.  Please create an issue with details and link it to this PR.  However I don't think this will help, and regardless will need to be undone during a coming refactor with internal IP addresses and we are moving off of the subnet you added since its a public IP block.  Please make an issue https://github.com/dappnode/DAppNodePackage-bitcoin/issues and explain in more detail your issue because just allowing the ip range wont help since you need to have an exposed port to the host. not to the internal docker network.

**Review by alexpeterson91 — CHANGES_REQUESTED** [review]
Please remove the version bump it will be done in another PR with proper versioning.
And create an issue for the exposable RPC part.

File: dappnode_package.json (hunk 1 above)
Comment (alexpeterson91) [inline dappnode_package.json:3]: revert so we can do version bump properly and apply semver in another PR, the core team will handle it.

File: dappnode_package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
+  "upstreamVersion": "v23.0",
```
Comment (alexpeterson91) [inline dappnode_package.json:4]: revert so we can do a version bump properly in its own PR

File: docker-compose.yml [hunk 4]
```diff
@@ -1,11 +1,11 @@
 version: "3.4"
 services:
   bitcoin.dnp.dappnode.eth:
-    image: "bitcoin.dnp.dappnode.eth:0.1.6"
+    image: "bitcoin.dnp.dappnode.eth:0.2.0"
```
Comment (alexpeterson91) [inline docker-compose.yml:4]: revert so we can do version bump properly and apply semver in another PR, the core team will handle it.

File: docker-compose.yml [hunk 5]
```diff
@@ -1,11 +1,11 @@
...
     build:
       context: ./build
       args:
-        UPSTREAM_VERSION: v22.0
+        UPSTREAM_VERSION: v23.0
```
Comment (alexpeterson91) [inline docker-compose.yml:8]: revert so we can do a version bump properly in its own PR

**Review by alexpeterson91 — CHANGES_REQUESTED** [review]
Please revert your version bump commit(s)

And for your RPC issue you need to create an issue here first github.com/dappnode/DAppNodePackage-bitcoin/issues with more context, you allowing RPC from the internal docker network will definitely not fix your issue, you need the container port 8332 to be forwarded (TCP) to a host port.  then from the local network network navigate/point to the LAN address of the dappnode followed by `:8332` if you set the ports the same will allow you RPC access to bitcoin otherwise use whatever host port docker forwards the bitcoin container port 8332 to.

**Comment by tropicar** [comment]
As @alexpeterson91 said, if we want to implement this feature, you should create an issue explaining why we would have to add these options to the package. We need that information in order to decide if it is a good idea to do it in a package by default that everybody will use. Thank you for the feedback!

File: docker-compose.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Leibniz137) [reply docker-compose.yml:19]: yeah, I also expose the port in the network settings via the dappnode gui (on the package's port settings page)

I am doing this because I need a connection to the bitcoin rpc port that is reliable for multiple days. I have always had trouble having uninterrupted connections to both the dappnode vpn as well as the dappnode wifi. (In the first case, my vpn connection will occasionally fail, and in the second case, the wifi stops being connected to the internet)

So instead of being on the vpn or dappnode wifi, I just stay on my normal wifi (which uses the normal 192.168.0.0/24 subnet) and assign my dappnode a private static ip address via my home router.


Here's my config sparrow config:
image

(the proxy config is irrelevant here)


I think that if you are going to set this to something other than `0.0.0.0/0` then you should allow the user to override the configuration. In that case you just need to add the environment variable to the docker-compose file

**Comment by Leibniz137** [comment]
reverted the various version changes so it's just the single feature.

I will open an issue and reference it here. I think it's pretty reasonable change, to keep whatever you want for that value as the default, but just allow the user to change it if they want to in the config section of the package.

File: docker-compose.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexpeterson91) [reply docker-compose.yml:19]: This one still breaks compatibility with other packages like LN I think need confirm from @tropicar also if you're on the same LAN this is unnecessary. Just forward the RPC port to the host from the docker container. You can then point it at the forwarded RPC port on the dappnode using the DAppNodes LAN address. You don't need to use a VPN or the dappnode Wi-Fi to connect like that if the port is properly exposed to the host. No need for proxies and changing the code of the package.

**Comment by alexpeterson91** [comment]
Just gotta remember to change this when we change this too when we do the docker subnet
